src/source.cpp

```diff
@@ -0,0 +1,84 @@
+// Source.cpp
+// Central Source Class
+
+#include <iomanip>
+#include <string>
+#include <vector>
+
+#ifndef PYNE_IS_AMALGAMATED
+#include "particle.h"
+#include "source.h"
+#endif
+
+/***************************/
+/*** Protected Functions ***/
+/***************************/
+
+// there are no protected functions currently
+// fool.
+
+/************************/
+/*** Public Functions ***/
+/************************/
+
+/*--- Constructors ---*/
+
+pyne::Source::Source() {}
+
+// Destructor
+pyne::Source::~Source() {}
+
+std::ostream& operator<<(std::ostream& os, pyne::Source tal) {
+  // print the Source to ostream
+  os << "\t---------\n";
+  return os;
+}
+
+pyne::PointSource::PointSource(double x, double y, double z, double i, double j,
+                               double k, double E, std::string p, double weight)
+    : x(x), y(y), z(z), i(i), j(j), k(k), E(E), particle(p), weight(weight) {}
```

Comment:
I'm used to seeing different parameters from the variables, either `_x(x)` or `x(_x)` or something like that, to be clear about scope.  I don't have a strong reference for that, though - is there guidance in the style guide?